Apps/W1/EDocumentConnectors/Tietoevry/app/src/Authenticator.Codeunit.al

```diff
@@ -0,0 +1,139 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.EServices.EDocumentConnector.Tietoevry;
+
+using System.Security.Authentication;
+codeunit 6394 "Authenticator"
+{
+    Access = Internal;
+    Permissions = tabledata "OAuth 2.0 Setup" = im,
+        tabledata "Connection Setup" = rim;
+
+    procedure CreateConnectionSetupRecord()
+    var
+        ConnectionSetup: Record "Connection Setup";
+    begin
+        if not ConnectionSetup.Get() then begin
```

Comment:
Thing to consider here:

Should it be possible to have 2 e-document services created, both using this connector? (prod, text) for instance. Or different formats?
Then it can be a problem with this single instance table.

Reply:
Not sure, how about Avalara?

Reply:
Similar issue in Avalara. We are looking at fixing it, as we probably need setup per service. But that is a requirement that comes from Avalara . So it depends on your service :)

Reply:
We plan only to have one format. There is no scenario when having one service to production and another to test.